provider/gandi/gandi.go

```diff
@@ -48,15 +49,49 @@ type GandiProvider struct {
 	LiveDNSClient LiveDNSClientAdapter
 	DomainClient  DomainClientAdapter
 	domainFilter  endpoint.DomainFilter
+	PreferALIAS   bool
 	DryRun        bool
+	TXTPrefix     string
+	TXTSuffix     string
 }
 
-func NewGandiProvider(ctx context.Context, domainFilter endpoint.DomainFilter, dryRun bool) (*GandiProvider, error) {
+const (
+	// RecordTypeALIAS is a RecordType enum value
+	RecordTypeALIAS = "ALIAS"
+)
+
+// inferZone determines the zone based on the RrsetName
+func inferZone(RrsetName string, TXTPrefix string, TXTSuffix string) string {
+	cleanRrsetName := strings.Replace(RrsetName, TXTPrefix, "", 1)
+	cleanRrsetName = strings.Replace(cleanRrsetName, TXTSuffix, "", 1)
+	cleanRrsetName = strings.Replace(cleanRrsetName, "cname-", "", 1)
+
+	return cleanRrsetName
+}
+
+// useAlias determines if the provider specific alias should be set
+func useAlias(domain string, preferALIAS bool) bool {
+	if preferALIAS {
+		return govalidator.IsDNSName(domain)
+	}
+
+	return false
+}
+
+// isGandiAlias determines if a given endpoint is supposed to create an Gandi Alias record
+func isGandiAlias(r livedns.DomainRecord, preferALIAS bool) bool {
+	return preferALIAS && r.RrsetType == RecordTypeALIAS
+}
+
+func NewGandiProvider(ctx context.Context, domainFilter endpoint.DomainFilter, dryRun bool, TXTPrefix string, TXTSuffix string) (*GandiProvider, error) {
 	key, ok := os.LookupEnv("GANDI_KEY")
+
 	if !ok {
 		return nil, errors.New("no environment variable GANDI_KEY provided")
 	}
+
 	sharingID, _ := os.LookupEnv("GANDI_SHARING_ID")
+	_, PreferALIAS := os.LookupEnv("GANDI_PREFER_ALIAS")
```

Comment:
Why are we taking configuration from environment variables? Shouldn't we be using the flags mechanism that everything else uses?

Reply:
The "PREFER_ALIAS" config was added to be "retro-compatible" with previous behaviour - which is to fail and raise an error. If it's alright we'd like to get rid of this config altogether and use the "CNAME is an ALIAS in gandi" feature by default
(I'll be working on this PR with @tommy31 - thanks for your review!)

Reply:
Updated in 2892142e13d6c50a469300616070683db6e99cdd - the fixed behaviour is now the default, and we stopped using an extra environment variable for this